llvm/test/CodeGen/SystemZ/atomic-store-06.ll

```diff
@@ -1,12 +1,11 @@
-; Test float atomic loads.
+; Test float atomic stores.
 ;
 ; RUN: llc < %s -mtriple=s390x-linux-gnu | FileCheck %s
 
 define void @f1(ptr %src, float %val) {
 ; CHECK-LABEL: f1:
-; CHECK: lgdr [[R:%r[0-9]+]], %f0
-; CHECK: srlg [[R]], [[R]], 32
-; CHECK: st [[R]], 0(%r2)
+; CHECK: ste %f0, 0(%r2)
+; CHECK: bcr 15, %r0
```

Comment:
Was the brc 15 missing before, or was just the check missing?  I see the check is also missing in the double version of this test.  These test should probably best be fully auto-generated.

Also, why do the long double versions of the tests use library calls?  Shouldn't they use lpq/stpq plus GPR/FPR moves?   In addition, for the long double (and also the i128) versions, we really should test both z10 and z13.

Reply:
Yes, the CHECK for bcr 15 was missing before - sorry should have mentioned that.

For the long double tests, see https://github.com/llvm/llvm-project/pull/75630#issuecomment-1861290278. I think the way the tests were written the output would actually be the same for z10/z13...